Source/NSubstitute/Core/CallResults.cs

```diff
@@ -6,12 +6,18 @@ namespace NSubstitute.Core
     public class CallResults : ICallResults
     {
         readonly ICallInfoFactory _callInfoFactory;
+        private readonly bool _skipVoidMethods;
         ConcurrentQueue<ResultForCallSpec> _results;
 
-        public CallResults(ICallInfoFactory callInfoFactory)
+        public CallResults(ICallInfoFactory callInfoFactory, bool skipVoidMethodsForPerformance)
```

Comment:
> Deeply inside I still don't agree with it and for me there is no difference (out is essentially the same return

I agree with you on that. Yes, `out` is a "return". NSub's API respects that. But it's internals were designed in a different way. And I really don't want to leak that exception tiny via protected method to a 3rd party library.

>I'll mention you and will ask for your review when I implement integration.
Yes please. That's not about horrible decisions 😆